File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc,string argv[])
{
    //making sure a key is given
    if (argc == 2)
{    
        int k = atoi(argv[1]);
        printf("plaintext:");
        string t = get_string();
        printf("ciphertext:");
        
        for ( int i = 0; i < strlen(t); i++)
        {
            //checking whether value is an alphabet
        if (isalpha(t[i]))    
        {    
            //checking whether alphabet is uppercase
            if (isupper(t[i])) 
            {
                printf("%c",(char) ((t[i] - 65) + k) % 26 + 65);
            } 
            //checking whether alphabet is lowercase
             else if (islower(t[i]))
            { 
                printf("%c",(char) ((t[i] - 97) + k) % 26 + 97);
            }
        }    
            else
            {
            printf("%c",t[i]);
            }
           
        }
    
}
    
    else 
    {
        printf("Error, please enter a key\n");
        return 1;  
    }
             printf("\n");
}

File: caesar.py
# Enciphers given text using Caesar's cipher
from cs50 import get_string
import sys


while True:
        key = int(sys.argv[1])
        if key > 0:
            break
ptext = get_string("Plaintext: ")
print ("ciphertext: ", end="")
    
for i in ptext:
 
    # Character is uppercase
    if i.isupper():
        print (chr((ord(i) - ord('A') + key) % 26 + ord('A')), end="")

    # Character is lowercase
    elif i.islower():
        print (chr((ord(i) - ord('a') + key) % 26 + ord('a')), end="")
 
    else:
        print(i, end="")
            
print()
    

                
            
    
    
    
    

